### PR TITLE
[MRG] Addressing Ch1 and Ch2 editorial reviews, add captions

### PR DESCRIPTION
**Pull request by hdashnow**
#233

### CODE REVIEW

File: markdown/ch1.markdown [hunk 1]
```diff
@@ -145,25 +151,29 @@ This is precisely the problem solved by the *NumPy array*.
...
-In our case we will need to store integers.
+Ndarrays underpin lots of awesome data manipulation techniques in SciPy.
+In particular, we're going to explore vectorization and broadcasting,
+techniques that allow us to write powerful, elegant code to manipulate our data.
 
+First, let's get our heads around the the ndarray.
+These arrays must be homogeneous; all items in an array must be the same type.
+In our case we will need to store integers.
 Ndarrays are called N-dimensional because they can have any number of dimensions.
 A 1-dimesional array is roughly equivalent to a Python list:
```
Comment (jni) [inline markdown/ch1.markdown]: ooh, typo-catch! dimesional->dimensional

File: markdown/ch1.markdown [hunk 2]
```diff
@@ -530,12 +544,16 @@ print('Count statistics:\n  min:  {0}\n  mean: {1}\n  max: {2}'
               np.max(total_counts)))
 ```
 
-<!-- caption text="Counts distribution" -->
+<!-- caption text="Density plot of gene expression counts per individual using KDE smoothing." -->
```
Comment (jni) [inline markdown/ch1.markdown]: No full stop at the end of captions (I checked other O'Reilly books for this)

File: markdown/ch1.markdown [hunk 3]
```diff
@@ -173,18 +183,20 @@ That will work, but it doesn't extend to *two-dimensional* arrays.
...
-print(two_d_array.shape)
+array2d = np.array(expression_data)
+print(array2d.shape)
 ```
 
 Now you can see that the `shape` attribute generalises `len` to account for the size of multiple dimensions of an array of data.
 
 ![multi-dimensional array diagram](../figures/NumPy_ndarrays.png)
 
+<!-- caption text="Visualizing NumPy's ndarrays in one, two and three dimensions." -->
```
Comment (jni) [inline markdown/ch1.markdown]: No full stop here either.

File: markdown/ch1.markdown [hunk 4]
```diff
@@ -1056,6 +1098,8 @@ ax.set_ylabel('log gene expression counts over all samples')
 plt.show()
 ```
 
+<!-- caption text="Comparing expression of two genes before RPKM normalization." -->
```
Comment (jni) [inline markdown/ch1.markdown]: No full stop

**Comment by hdashnow** [comment]
BTW, it looked like the ch2 editorial review was partially addressed. I added some subheadings in different spots from the recommended, but I think they work better there.

**Comment by jni** [comment]
@hdashnow it's possible that one of us addressed it ages ago. We got these more than 6 months ago now!

**Pull request merged**